sdk/core/core-http/src/serializer.ts

```diff
@@ -697,21 +697,28 @@ function deserializeCompositeType(
         );
       } else {
         const propertyName = xmlElementName || xmlName || serializedName;
-        let unwrappedProperty = responseBody[propertyName!];
         if (propertyMapper.xmlIsWrapped) {
-          unwrappedProperty = responseBody[xmlName!];
-          unwrappedProperty = unwrappedProperty && unwrappedProperty[xmlElementName!];
-
-          const isEmptyWrappedList = unwrappedProperty === undefined;
-          if (isEmptyWrappedList) {
-            unwrappedProperty = [];
-          }
+          /* a list of <xmlElementName> wrapped by <xmlName>
+            For the xml example below
+              <Cors>
+                <CorsRule>...</CorsRule>
+                <CorsRule>...</CorsRule>
+              </Cors>
+            the responseBody has
+              {
+                Cors: {
+                  CorsRule: [{...}, {...}]
+                }
+              }
+            xmlName is "Cors" and xmlElementName is"CorsRule".
+          */
+          const wrapped = responseBody[xmlName!];
+          const elementList = wrapped?.[xmlElementName!] ?? [];
+          instance[key] = serializer.deserialize(propertyMapper, elementList, propertyObjectName);
+        } else {
+          const property = responseBody[propertyName!];
+          instance[key] = serializer.deserialize(propertyMapper, property, propertyObjectName);
```

Comment:
Do we already have a test case that covers the else block? If not would it be possible to add one?

Reply:
Yes, other xml tests don't have `xmlIsWrapped: true` so the else branch is exercised. Code coverage result also confirmed that.  (Now I see those tests in deserializationPolicyTests.ts I should probably move these new tests there.)

Reply:
Great!!! Thanks